web/packages/teleterm/src/ui/services/connectionTracker/connectionTrackerService.ts

```diff
@@ -220,6 +231,23 @@ export class ConnectionTrackerService extends ImmutableStore<ConnectionTrackerSt
             }
             break;
           }
+          case 'doc.gateway_kube': {
+            const gwConn = draft.connections.find(
+              getGatewayKubeConnectionByDocument(doc)
+            ) as TrackedGatewayKubeConnection;
+
+            if (!gwConn) {
+              const newItem = createGatewayKubeConnection(doc);
+              draft.connections.push(newItem);
+            } else {
+              gwConn.gatewayUri = doc.gatewayUri;
+              gwConn.port = doc.port;
+              gwConn.connected = !!this._clusterService.findGateway(
+                doc.gatewayUri
```

Comment:
After removing `gatewayUri` from the new doc type, we can use `ClustersService.findGatewayByConnectionParams` instead, similar to how we do this in `useDocumentTerminal`.